base/reflection.jl

```diff
@@ -126,7 +126,12 @@ julia> fieldname(SparseMatrixCSC, 5)
 :nzval
 ```
 """
-fieldname(t::DataType, i::Integer) = t.name.names[i]::Symbol
+function fieldname(t::DataType, i::Integer)
+    n_fields = length(t.name.names)
+    field_label = n_fields == 1 ? "field" : "fields"
+    i > n_fields && error("Cannot access field $i since type $t only has $n_fields $field_label.")
```

Comment:
This would probably be better as `throw(ArgumentError("..."))` rather than `error("...")`. Also a `@test_throws` for this case would be good.